contracts/factory/MagicDropCloneFactory.sol

```diff
@@ -0,0 +1,256 @@
+        _initializeOwner(initialOwner);
+
+        emit MagicDropFactoryInitialized();
+    }
+
+    /*==============================================================
+    =                      PUBLIC WRITE METHODS                    =
+    ==============================================================*/
+
+    /// @notice Creates a new deterministic clone of a MagicDrop contract
+    /// @param name The name of the new contract
+    /// @param symbol The symbol of the new contract
+    /// @param standard The token standard of the new contract
+    /// @param initialOwner The initial owner of the new contract
+    /// @param implId The implementation ID
+    /// @param salt A unique salt for deterministic address generation
+    /// @return The address of the newly created contract
+    function createContractDeterministic(
+        string calldata name,
+        string calldata symbol,
+        TokenStandard standard,
+        address payable initialOwner,
+        uint32 implId,
+        bytes32 salt
+    ) external returns (address) {
+        // Retrieve the implementation address from the registry
+        (address impl, bool deprecated) = _registry.getImplementation(standard, implId);
+
+        if (deprecated) {
+            revert ImplementationDeprecated();
+        }
+
+        if (impl == address(0)) {
+            revert ImplementationNotRegistered();
+        }
+
+        /// @solidity memory-safe-assembly
+        assembly {
+            mstore(0x00, salt)
+            mstore(0x20, MAGICDROP_FACTORY_STORAGE)
+            let saltUsed := sload(keccak256(0x00, 0x40)) // usedSalts[salt]
+
+            if saltUsed {
+                mstore(0x00, 0x0ced3043) // SaltAlreadyUsed()
+                revert(0x1c, 0x04)
+            }
+        }
+
+        // Predict the address where the contract will be deployed
+        address predictedAddress = LibClone.predictDeterministicAddress(impl, salt, address(this));
+
+        // Check if a contract already exists at the predicted address
+        if (predictedAddress.code.length > 0) {
+            revert ContractAlreadyDeployed(predictedAddress);
+        }
+
+        // Create a deterministic clone of the implementation contract
+        address instance = LibClone.cloneDeterministic(impl, salt);
+
+        // Initialize the newly created contract
+        (bool success,) = instance.call(abi.encodeWithSelector(INITIALIZE_SELECTOR, name, symbol, initialOwner));
+        if (!success) {
+            revert InitializationFailed();
+        }
+
+        /// @solidity memory-safe-assembly
+        assembly {
+            mstore(0x00, salt)
+            mstore(0x20, MAGICDROP_FACTORY_STORAGE)
+            sstore(keccak256(0x00, 0x40), 1) // usedSalts[salt] = true
+        }
+
+        emit NewContractInitialized({
+            contractAddress: instance,
+            initialOwner: initialOwner,
+            implId: implId,
+            standard: standard,
+            name: name,
```

Comment:
<img width="1396" alt="Screenshot 2024-10-08 at 9 50 55 PM" src="https://github.com/user-attachments/assets/b8fea397-1ab9-40d3-af98-737fc987af07">


I get a stack too deep error when running `forge build` 

bypassed by adding `via_ir = true` to `foundry.toml`

Reply:
yeah im aware of this, i can add via_ir to the foundry, though i was hoping future refactors can fix this issue all together. i think this must be set in hardhat already